### ISSUE TITLE
[Multi User] Integrate KFP multi user with KF 1.1

### ISSUE THREAD

**Issue by Bobgy** [kubeflow/pipelines#3241]
Part of https://github.com/kubeflow/pipelines/issues/1223 (that's the main issue tracker).

I'm using this issue as a tracker for all the efforts of merging KFP multi user related changes to upstream kubeflow repo.

/priority p0
/assign @Bobgy 
/kind feature

My fork to merge kfp multi user manifests with KF 1.0: https://github.com/Bobgy/manifests/tree/kfp-multi-user-master

* [x] Enable istio-injection in kubeflow namespace. [kubeflow#3866](https://github.com/kubeflow/kubeflow/issues/3866) blocked by
  * [x] Two examples for disabling istio injection in kubeflow namespace kubeflow/manifests#703
  * [x] Add sidecar.istio.io/inject false annotations to all stateful sets in kubeflow ns kubeflow/manifests#711
  * [x] Add istio sidecar inject false to deployments in kubeflow ns kubeflow/manifests#712
  * [x] Added istio sidecar false annotation to metadata deployments https://github.com/kubeflow/manifests/pull/804
  * [x] [Backend] Always disable istio sidecar injection https://github.com/kubeflow/pipelines/pull/3305
  * [x] Disable istio sidecar injection in simple tfjob https://github.com/kubeflow/tf-operator/pull/1148
  * [x] https://github.com/kubeflow/kfctl/issues/296
  * [x] Add kubeflow namespace resource to enable istio injection kubeflow/manifests#1153
* [ ] Secure KFAM with mutual TLS and istio authorization rules https://github.com/kubeflow/kubeflow/issues/4871 blocked by
  * [x] [KFAM] LivenessProbe doesn’t work with istio mutual TLS [kubeflow#4865](https://github.com/kubeflow/kubeflow/issues/4865)
  * [x] Profiles deployment has missing port [kubeflow/manifest#976](https://github.com/kubeflow/manifests/issues/976) Port is required for istio sidecar to intercept traffic for the pod.
* [x] Migrate Kubeflow's KFP manifests using kustomize v3
  * [x] Argo kustomize v3 https://github.com/kubeflow/manifests/issues/1090
  * [x] KFP kustomize v3 https://github.com/kubeflow/manifests/issues/1048
* [x] Enable multi user mode for Kubeflow Pipelines https://github.com/kubeflow/pipelines/issues/4159 https://github.com/kubeflow/pipelines/issues/4160 https://github.com/kubeflow/gcp-blueprints/pull/80
* [x] Some last mile bug fixes https://github.com/kubeflow/manifests/pull/1374
* [ ] Integrate with non GCP platforms https://github.com/kubeflow/manifests/issues/1364

**Comment by nrchakradhar** [comment]
@Bobgy Though this is not strictly multi-user, https://github.com/kubeflow/pipelines/issues/2858, this needs to be addressed since users may want to use different namespace.

**Comment by Bobgy** [comment]
@nrchakradhar Thanks for bringing this up, we have addressed it in https://github.com/kubeflow/pipelines/pull/2665 (but we are putting together all pieces of multi user support, you cannot use it right now). This issue is meant to track efforts that need to be merged into Kubeflow, so this particular issue isn't tracked here.

**Comment by nrchakradhar** [comment]
@Bobgy Thanks for the reference.

**Comment by Jeffwan** [comment]
@Bobgy Do we have a timeline or milestone when multi-user in pipeline will be available?

**Comment by Bobgy** [comment]
Hi @Jeffwan! Please see the main issue tracker for progress: https://github.com/kubeflow/pipelines/issues/1223.
And you may try phase 1 multi user support in https://github.com/kubeflow/pipelines/issues/1223#issuecomment-603675906
(We are working on phase 2)

**Comment by Jeffwan** [comment]
> Hi @Jeffwan! Please see the main issue tracker for progress: #1223.
> And you may try phase 1 multi user support in [#1223 (comment)](https://github.com/kubeflow/pipelines/issues/1223#issuecomment-603675906)
> (We are working on phase 2)

Cool. I will have a check on phase I first and leave some feedbacks

**Comment by jlewi** [comment]
@Bobgy Could you please clarify what this issue is tracking? What is the remaining work? You asked me https://github.com/kubeflow/pipelines/issues/1223#issuecomment-630509920
to take a look. What is it you want me to take a look at?

**Comment by Bobgy** [comment]
@jlewi This tracks remaining work to merge manifests to kubeflow manifests master. The unchecked items above are remaining. (I guess you'd want a breakdown for Enable multi user mode for Kubeflow Pipelines, I'll keep updating this when working on this.)

**Comment by Bobgy** [comment]
I'm now using https://github.com/kubeflow/pipelines/projects/5 to track efforts merging multi user mode to KF 1.1. Closes this.

**Comment by Bobgy** [comment]
In fact, this issue is still clearer on the steps. Keep using this one

**Comment by Bobgy** [comment]
TODO: I also need to cherry pick a bunch of multi user mode PRs to 1.1 branch.

**Comment by jlewi** [comment]
@Bobgy Do we have an ETA for when all of the PRs will be on the 1.1 branch?

**Comment by Bobgy** [comment]
@jlewi https://github.com/kubeflow/gcp-blueprints/pull/80 and https://github.com/kubeflow/manifests/pull/1374 are the only ones left.
If they are merged, I can create cherry pick PRs tomorrow.

**Comment by jlewi** [comment]
kubeflow/gcp-blueprints#80 Per comment in this issue turning on istio-sidecar injection in the kubeflow namespace is a pretty big change. Do we really need to it this late in the release cycle?

**Comment by Bobgy** [comment]
@jlewi oooops, I just realized ASM 1.4 doesn't support RBAC v1alpha1: https://cloud.google.com/service-mesh/docs/archive/1.4/docs/supported-features#authorization_policy.

I'll port the istio RBAC policies to Authorization v1beta1

**Comment by Bobgy** [comment]
It will be the same problem for any component that use `servicerole` and `servicerolebinding`s to authorize like jupyter notebooks.

**Comment by Bobgy** [comment]
> It will be the same problem for any component that use `servicerole` and `servicerolebinding`s to authorize, e.g. jupyter notebooks.

Hmmmm, I was wrong. I tried a notebook, when accessing it from another account without permission. I got `RBAC: access denied` as expected. So something is not working correctly for KFP. I'll troubleshoot it.

**Comment by Bobgy** [comment]
OK, I found the culprit, it's knative's istio rule: https://github.com/kubeflow/manifests/blob/c300ee9ca30d8b606cd9713d120721f36e86b0ec/knative/knative-serving-install/base/service-role-binding.yaml#L11.

It basically allowed any network access from any user to any services in the namespace knative was deployed to.

But knative was already moved to its own namespace, so we can test latest version.

**Comment by Bobgy** [comment]
/cc @jlewi 
I found the cause, see above comment.

**Comment by Bobgy** [comment]
I'll keep this open until verified on GCP platform

**Comment by Bobgy** [comment]
UPDATE: verified KFP 1.0.0 with multi user mode working properly on GCP platform.
